flexfactor_tenets_tests.py

```
@@ -0,0 +1,375 @@
+        with mock.patch.object(
+            ft,
+            "cached_tenets_context",
+            return_value=self._result("src/target.py"),
+        ):
+            ft.install(runtime)
+            result = runtime["_enumerate_source_files"](str(self.root), 2)
+        self.assertEqual(result, ["src/target.py", "src/first.py"])
+
+    def test_ranked_file_beyond_global_cap_enters_budget_and_cap_is_restored(self) -> None:
+        runtime: dict[str, object] = {"MAX_FILES_PER_RUN": 2}
+        exec(
```

Ruff (0.16.3)

[error] Use of `exec` detected

(S102)